pkg/services/pluginsintegration/clientmiddleware/caching_middleware.go

```diff
@@ -77,7 +90,12 @@ func (m *CachingMiddleware) QueryData(ctx context.Context, req *backend.QueryDat
 
 	// Update the query cache with the result for this metrics request
 	if err == nil && cr.UpdateCacheFn != nil {
-		cr.UpdateCacheFn(ctx, resp)
+		// If AWS async cahing is enabled and it is a still running async query, don't cache it
```

Comment:
```suggestion
		// If AWS async caching is enabled and it is a still running async query, don't cache it
```

Reply:
If async queries are disabled, we always want to call the function. If they are enabled, we only call the function conditionally. The logic is correct but maybe the comment could be a tad clearer